scripts/check_compliance.py

```diff
@@ -895,211 +891,225 @@ def init_logs(cli_arg):
+                if msg in comment.body:
+                    return comment
+
+        return None
+
+    if cached_bot_comment == 0:
+        cached_bot_comment = get_comment()
+    return cached_bot_comment
 
-    return comment_count
+
+# Cache used by get_bot_comment(). Use 0 instead of None for "no cached value"
+# so that None (no comment) can be cached.
+cached_bot_comment = 0
 
 
 def parse_args():
-    """
-    Parse arguments
-    :return:
-    """
     parser = argparse.ArgumentParser(
         description="Check for coding style and documentation warnings.")
     parser.add_argument('-c', '--commits', default="HEAD~1..",
                         help='''Commit range in the form: a..[b], default is
                         HEAD~1..HEAD''')
     parser.add_argument('-g', '--github', action="store_true",
-                        help="Send results to github as a comment.")
+                        help="Post results as comments in the PR on GitHub")
 
     parser.add_argument('-r', '--repo', default=None,
-                        help="Github repository")
+                        help="GitHub repository")
     parser.add_argument('-p', '--pull-request', default=0, type=int,
                         help="Pull request number")
 
     parser.add_argument('-s', '--status', action="store_true",
-                        help="Set status to pending")
+                        help="Set status on GitHub to pending and exit")
     parser.add_argument('-S', '--sha', default=None, help="Commit SHA")
     parser.add_argument('-o', '--output', default="compliance.xml",
                         help='''Name of outfile in JUnit format,
                         default is ./compliance.xml''')
 
     parser.add_argument('-l', '--list', action="store_true",
-                        help="List all test modules.")
+                        help="List all checks and exit")
 
     parser.add_argument("-v", "--loglevel", help="python logging level")
 
     parser.add_argument('-m', '--module', action="append", default=[],
-                        help="Test modules to run, by default run everything.")
+                        help="Checks to run. All checks by default.")
 
     parser.add_argument('-e', '--exclude-module', action="append", default=[],
-                        help="Do not run the specified modules")
+                        help="Do not run the specified checks")
 
     parser.add_argument('-j', '--previous-run', default=None,
                         help='''Pre-load JUnit results in XML format
                         from a previous run and combine with new results.''')
 
-
     return parser.parse_args()
 
 
-def main():
-    """
-    Main function
+def init_github(args):
+    # Initializes a GitHub connection, if needed
 
-    :return:
-    """
+    global commit_sha
+    global github_repo
+    global github_pr
+    global github_commit
+
+    if not (args.github or args.status):
+        # No GitHub-related stuff to do
+        return
```

Comment:
I think it would be slightly more readable to have this test in main().

Extracting github stuff most users never run into a different script would also make this one easier to read and maintain however. 1. I don't if that's realistic 2. I digress.

Reply:
Yeah, could change it later maybe.